tests/functional/backend/wmg/test_wmg_api.py

```diff
@@ -0,0 +1,80 @@
+        """
+        MAX_RESPONSE_TIME_SECONDs = 1.5
+        start_time = time.time()
+        res = requests.get(f"{self.api}/primary_filter_dimensions")
+        end_time = time.time()
+        func_call_time_ms = end_time - start_time
+        self.assertGreater(MAX_RESPONSE_TIME_SECONDs, func_call_time_ms)
+        self.assertEqual(res.status_code, requests.codes.ok)
+
+    def test_secondary_filters_common_case(self):
+        """
+        1 tissue w/50 cell types, 20 genes, 3 secondary filters specified
+        Returns in less than 2 seconds
+        """
+        MAX_RESPONSE_TIME_SECONDS = 2
+        headers = {"Content-Type": "application/json"}
+        data = {"filter": {"dataset_ids": [],
+                           "disease_ontology_term_ids": ["MONDO:0005812", "MONDO:0100096", "PATO:0000461"],
+                           "ethnicity_ontology_term_ids": ["HANCESTRO:0010", "HANCESTRO:0014"],
+                           "gene_ontology_term_ids": list(genes_20_count.keys()),
+                           "organism_ontology_term_id": "NCBITaxon:9606",
+                           "sex_ontology_term_ids": ["PATO:0000383"],
+                           "tissue_ontology_term_ids": ["UBERON:0000178"]},  # blood (more than 50 cell types)
+                "include_filter_dims": True,
+                "snapshot_id": self.data["snapshot_id"]}
+        start_time = time.time()
+        res = requests.post(f"{self.api}/query", data=json.dumps(data), headers=headers)
+        end_time = time.time()
+        func_call_time_ms = end_time - start_time
+        self.assertGreater(MAX_RESPONSE_TIME_SECONDS, func_call_time_ms)
+        self.assertEqual(res.status_code, requests.codes.ok)
+
+    def test_secondary_filters_extreme_case(self):
+        """
+        4 tissues w/largest cell type counts, 400 genes, no secondary filtering
+        Returns in less than 10 seconds
+        """
+        MAX_RESPONSE_TIME_SECONDS = 10
+        headers = {"Content-Type": "application/json"}
+        data = {"filter": {"dataset_ids": [],
+                           "disease_ontology_term_ids": [],
+                           "ethnicity_ontology_term_ids": [],
+                           "gene_ontology_term_ids": list(genes_400_count.keys()),
+                           "organism_ontology_term_id": "NCBITaxon:9606",
+                           "sex_ontology_term_ids": [],
+                           "tissue_ontology_term_ids": ["UBERON:0000178", "UBERON:0002048", "UBERON:0000029",
+                                                        "UBERON:0000970", "UBERON:0000362"]},
+                "include_filter_dims": True,
+                "snapshot_id": self.data["snapshot_id"]}
+        start_time = time.time()
+        res = requests.post(f"{self.api}/query", data=json.dumps(data), headers=headers)
+        end_time = time.time()
+        func_call_time_ms = end_time - start_time
+        self.assertGreater(MAX_RESPONSE_TIME_SECONDS, func_call_time_ms)
```

Comment:
might also want to assert result size is "reasonably" large, and hasn't shrunk due to changed or bad data. This is more of a sanity check the func test itself is still valid, and not something the test is trying validate directly.

Reply:
I added a check that the content isnt empty, but "reasonably large" is hard to define given that it will be different based on the environment/cube